src/mount/writedata.cc

```diff
@@ -256,42 +255,32 @@ void write_cb_wait_for_block(inodedata* id, Glock& glock) {
 
 /* inode */
 
-inodedata* write_find_inodedata(uint32_t inode, Glock&) {
-	uint32_t idh = IDHASH(inode);
-	for (inodedata* id = idhash[idh]; id; id = id->next) {
-		if (id->inode == inode) {
-			return id;
-		}
+inodedata *write_find_inodedata(uint32_t inode, Glock &) {
+	if (inodedataMap.contains(inode)) {
+		return inodedataMap[inode];
 	}
 	return NULL;
 }
 
-inodedata* write_get_inodedata(uint32_t inode, Glock&) {
-	uint32_t idh = IDHASH(inode);
-	inodedata* id;
-	for (inodedata* id = idhash[idh]; id; id = id->next) {
-		if (id->inode == inode) {
-			return id;
-		}
+inodedata *write_get_inodedata(uint32_t inode, Glock &) {
+	if (inodedataMap.contains(inode)) {
+		return inodedataMap[inode];
 	}
-	id = new inodedata(inode);
-	id->next = idhash[idh];
-	idhash[idh] = id;
+	auto id = new inodedata(inode);
+	inodedataMap[inode] = id;
 	return id;
 }
 
-void write_free_inodedata(inodedata* fid, Glock&) {
-	uint32_t idh = IDHASH(fid->inode);
-	inodedata *id, **idp;
-	idp = &(idhash[idh]);
-	while ((id = *idp)) {
-		if (id == fid) {
-			*idp = id->next;
-			delete id;
-			return;
-		}
-		idp = &(id->next);
+void write_free_inodedata(inodedata *fid, Glock &) {
+	uint32_t inode = fid->inode;
+	if (!inodedataMap.contains(inode)) {
+		return;
 	}
+
+	auto id = inodedataMap[inode];
+    sassert(id == fid);
```

Comment:
Check the indentation of this line. It looks different.

Reply:
Applied.